src/CMakeLists.txt

```diff
@@ -83,6 +83,9 @@ ENDIF()
 
 add_executable(NimbleSM_Serial nimble_serial_main.cc)
 target_link_libraries(NimbleSM_Serial PRIVATE nimble::nimble)
+if (NIMBLE_HAVE_KOKKOS)
+  target_link_libraries(NimbleSM_Serial PRIVATE Kokkos::kokkos)
```

Comment:
One thing we could consider is refactoring this file a bit and doing target_link_library(nimble PUBLIC Kokkos::kokkos) earlier (in the root CMakeLists.txt I think this would mean less special cases (cmake does transitive linking so making the base library link to kokkos means all executables will link to it). For now I think this is good, just something to consider for the future